src/main/java/DateTime.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Represents a date and time object to store and convert
+ * date and time to a pleasant format e.g., 12/11/2019 1700
+ * becomes 12th November 2019, 5:00pm.
+ */
+public class DateTime {
+    private String date;
+    private String time;
+
+    public DateTime(String date, String time) {
+        this.date = date;
+        this.time = time;
+    }
+
+    public DateTime(String dateTime) throws InvalidInputException {
+        String[] splitDateAndTime = dateTime.split(" ");
+        String[] splitDate = splitDateAndTime[0].split("/");
+        String day = splitDate[0];
+        String month = splitDate[1];
+        String year = splitDate[2];
+
+        if (day.equals("1") || day.equals("21") || day.equals("31")) {
+            day = day + "st";
+        } else if (day.equals("2") || day.equals("22")) {
+            day = day + "nd";
+        } else if (day.equals("3") || day.equals("23")) {
+            day = day + "rd";
+        } else {
+            day = day + "th";
+        }
+
+        switch (month) {
```

Comment:
I would like to suggest using the Java 8 API "java.time" to parse the date string to check if the date is valid and get the date string in return. By parsing the date string using a date time format, it will check if the date string is in a valid format, otherwise, it will throw an error.
Example to check if date string is valid:
> try {
            String dateStr = "32/12/2019 1800";
            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
            LocalDateTime dateTime = LocalDateTime.parse(dateStr, formatter);
        } catch (DateTimeParseException e) {
            System.err.println(e);
        }

Example to convert date to String format to save:

> LocalDateTime dateTime = this.getLocalDateTime();
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d MMMM yyyy, h:mma");
        String formattedDateTime = dateTime.format(formatter);

This way, you wouldn't have to check for the date time format manually. You can then add the suffix of the date string.